spring-tx/src/main/java/org/springframework/transaction/support/AbstractPlatformTransactionManager.java

```diff
@@ -1239,8 +1239,10 @@ protected void doSetRollbackOnly(DefaultTransactionStatus status) throws Transac
 	protected void registerAfterCompletionWithExistingTransaction(
 			Object transaction, List<TransactionSynchronization> synchronizations) throws TransactionException {
 
-		logger.debug("Cannot register Spring after-completion synchronization with existing transaction - " +
-				"processing Spring after-completion callbacks immediately, with outcome status 'unknown'");
+		if(logger.isDebugEnabled()){
+			logger.debug("Cannot register Spring after-completion synchronization with existing transaction - " +
+					"processing Spring after-completion callbacks immediately, with outcome status 'unknown'");
+		}
```

Comment:
No need for the `isDebugEnabled()` check since this is effectively a constant string. In other words, there is nothing dynamic about this string concatenation which should be converted to a single string by the compiler.

```suggestion
		logger.debug("Cannot register Spring after-completion synchronization with existing transaction - " +
				"processing Spring after-completion callbacks immediately, with outcome status 'unknown'");
```